File: src/main/java/kr/co/lunatalk/domain/product/dto/response/ProductCreateResponse.java
package kr.co.lunatalk.domain.product.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import kr.co.lunatalk.domain.product.domain.Product;

public record ProductCreateResponse(
	@Schema(description = "상품 ID")
	Long productId
) {
	public static ProductCreateResponse from(Product product) {
		return new ProductCreateResponse(product.getId());
	}
}

File: lunatalk-domain/src/main/kotlin/kr/co/lunatalk/domain/product/dto/response/ProductCreateResponse.kt
package kr.co.lunatalk.domain.product.dto.response

import io.swagger.v3.oas.annotations.media.Schema
import kr.co.lunatalk.domain.product.domain.Product

data class ProductCreateResponse(
    @Schema(description = "상품 ID")
    val productId: Long?
) {
    companion object {
        fun from(product: Product): ProductCreateResponse {
            return ProductCreateResponse(product.id)
        }
    }
}
